app/api/v1/endpoints/rate_limit_endpoints.py

```
@@ -0,0 +1,335 @@
+                    "pro_rpm": pro_rpm_usage,
+                    "pro_rpd": pro_rpd_usage
+                }
+            }
+        )
+        
+    except Exception as e:
+        logger.error(f"Failed to get rate limit status: {e}")
+        raise HTTPException(
+            status_code=500,
+            detail=f"Failed to retrieve rate limit status: {str(e)}"
+        )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)